assembl/static2/js/app/pages/brightMirrorFiction.jsx

```diff
@@ -164,24 +171,30 @@ export class BrightMirrorFiction extends Component<LocalBrightMirrorFictionProps
       });
   };
 
-  // Create an array of comments from the current fiction
+  // Fetch top comments and total number of comments from fictionId
   // The fiction is the main post, a post (or comment) with a parentId identical to the fictionId
   // will be considered as the top level of comments of the fiction
-  getTopComments() {
+  getCommentsInfo() {
     const { fictionId, ideaWithCommentsData } = this.props;
     const { edges } = ideaWithCommentsData.idea.posts;
     const { idea, refetch } = ideaWithCommentsData;
 
     if (!ideaWithCommentsData.idea) return [];
 
-    const topComments = head(
-      transformPosts(edges, [], {
-        refetchIdea: refetch,
-        ideaId: idea.id
-      }).filter(post => post.id === fictionId)
-    ).children;
+    const transformedPosts = transformPosts(edges, [], {
+      refetchIdea: refetch,
+      ideaId: idea.id
+    }).filter(post => post.id === fictionId);
+
+    const topComments = head(transformedPosts).children;
+    const commentsCount = getDebateTotalMessages(transformedPosts) - 1; // We remove 1 since one entry is the fiction post itself
```

Comment:
Isnt the fiction filtered above because you only take 'head(transformedPosts).children' of 'transformedPosts' which is the fiction ?

Reply:
you are so right... I'll fix that :)